gateway-server/src/main/java/org/apache/knox/gateway/webshell/JSchConnectInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.webshell;
+
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import org.eclipse.jetty.io.RuntimeIOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+/**
+ * data structure to store a connection session
+ */
+public class JSchConnectInfo extends ConnectInfo {
+
+    private JSch jsch;
+    private Channel channel;
+    private Session jschSession;
+    private static final Logger LOG = LoggerFactory.getLogger(JSchConnectInfo.class);
+    public JSchConnectInfo(){
+        try {
+            jsch = new JSch();
+            Properties config = new Properties();
```

Comment:
We cannot have hardcoded properties especially usernames and passwords :)

Reply:
ultimately, we want to remove the JSch connection and use Process connection instead, which wont require a hard coded username/password